### PR TITLE
Adding alerts stream in OpsGenie Connector

### PR DESCRIPTION
**Pull request by mudasirmirza**
## Description

> Provide description here

## Type of change
- [X] New feature

## Extra info

> Adding Alerts stream in OpsGenie connector

## Test Results

```
➜  opsgenie-source git:(mudasir/adding_alerts_stream_opsgenie) npm test

> opsgenie-source@0.0.1 test
> jest --verbose --color

 PASS  test/index.test.ts
  index
    ✓ spec (2 ms)
    ✓ check connection (1 ms)
    ✓ check connection - incorrect config (1 ms)
    ✓ streams - incidents, use full_refresh sync mode (2 ms)
    ✓ streams - incidents, paginate
    ✓ streams - teams, use full_refresh sync mode (1 ms)
    ✓ streams - users, use full_refresh sync mode (1 ms)
    ✓ streams - users, paginate
    ✓ streams - alerts, use full_refresh sync mode (1 ms)
    ✓ streams - alerts, paginate (1 ms)

Test Suites: 1 passed, 1 total
Tests:       10 passed, 10 total
Snapshots:   0 total
Time:        0.45 s, estimated 5 s
Ran all test suites.
➜  opsgenie-source git:(mudasir/adding_alerts_stream_opsgenie) 
```

### CODE REVIEW

**Comment by mudasirmirza** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by mudasirmirza** [comment]
recheck

File: sources/opsgenie-source/src/opsgenie/opsgenie.ts [hunk 1]
```diff
@@ -144,6 +145,29 @@ export class OpsGenie {
...
+
+    let offset = 0;
+    do {
+      const params = {offset, limit: this.pageSize, sort: 'createdAt'};
+      const response = await this.retryApi<PaginateResponse<Alert>>(
+        `v2/alerts`,
+        params
+      );
+      for (const alert of response?.data?.data ?? []) {
+        if (new Date(alert.createdAt) >= startTimeMax) {
```
Comment (tovbinm) [inline sources/opsgenie-source/src/opsgenie/opsgenie.ts:160]: I would change `>=` to `>`

File: sources/opsgenie-source/src/opsgenie/opsgenie.ts [hunk 2]
```diff
@@ -144,6 +145,29 @@ export class OpsGenie {
     } while (true);
   }
 
+  async *getAlerts(createdAt?: Date): AsyncGenerator<Alert> {
+    const startTimeMax =
+      createdAt > this.startDate ? createdAt : this.startDate;
+
+    let offset = 0;
+    do {
+      const params = {offset, limit: this.pageSize, sort: 'createdAt'};
```
Comment (tovbinm) [inline sources/opsgenie-source/src/opsgenie/opsgenie.ts:154]: do we check that `this.pageSize` is a positive integer number somewhere?

**Review by tovbinm — APPROVED** [review]
@mudasirmirza lgtm! see some minor comments

File: sources/opsgenie-source/src/opsgenie/opsgenie.ts (hunk 2 above)
Comment (mudasirmirza) [reply sources/opsgenie-source/src/opsgenie/opsgenie.ts:154]: @tovbinm this is not being checked in this connector anywhere.

Comment (tovbinm) [reply sources/opsgenie-source/src/opsgenie/opsgenie.ts:154]: Can you please add a check here ? https://github.com/faros-ai/airbyte-connectors/blob/main/sources/opsgenie-source/src/opsgenie/opsgenie.ts#L34

Comment (mudasirmirza) [reply sources/opsgenie-source/src/opsgenie/opsgenie.ts:157]: @tovbinm added a check to see if the provided page size value is a positive value or not

**Review by tovbinm — APPROVED** [review]

**Pull request merged**